### ISSUE TITLE
CRM integration - Salesforce not receiving content of Entries element field selection

### ISSUE THREAD

**Issue by j-greig** [verbb/formie#197]
We have an "Entries" field that lets you pick an entry from a section.

The selected entry shows correctly in the Submissions list in Craft, and all of the other form fields (which are text-based) are being received by Salesforce, but the Entries field shows up blank. 

Possibly related to https://github.com/verbb/formie/issues/196

**Comment by engram-design** [comment]
What field are you mapping your entries field to in Salesforce?

**Comment by j-greig** [comment]
Ahh good question, I'm not 100% sure as our client had already created a bunch of fields.... should we create a new one, and if so do you have any idea what type it should be? (sorry for being clueless about Salesforce)

**Comment by j-greig** [comment]
Update - mapped to a Text (100) field

**Comment by engram-design** [comment]
So the tricky part about an element field is what data should be used? The ID, Title, or URL of the element? What would you (or your client) expect as the desired outcome?

**Comment by j-greig** [comment]
In this scenario, just the title of the element - as we want the Salesforce entry to be a standalone record of the enquiry.

I won't say I'd expect that to always be the case, but I can't see many situations where someone would want to reverse engineer the connection and do something with the ID in Salesforce itself (?)

**Comment by engram-design** [comment]
Hmmm, so this poses a bit of a challenge for a few reasons. Firstly, element fields are natively multi-valued, so we're always dealing with array, and that's expected behaviour. Instead of a single value, which would be used in a text field.

Secondly, if you had an element field, you'd expect the entire element to be sent to the provider, rather than just the title. For example - with a Webhook integration, you'd want the full element information. There's also the issue with some integrations wanting different things.

But, I've managed to refactor things to address different destination field types for integrations.

In addition, I've added a `parseMappedFieldValue` event that allows you to parse content from Formie into the integration provider however you like!

See [1.2.25](https://github.com/verbb/formie/releases/tag/1.2.25)
